addons/capetrader/capetrader.lua

```diff
@@ -0,0 +1,495 @@
+    else
+        currentlyAugmenting = false
+    end
+end
+
+function checkAugLimits()
+    local capeItem
+    for index, item in pairs(inventory) do
+        if index ~= 'max' and index ~= 'count' and index ~= 'enabled' then
+            if item.id == currentCape.id then
+                capeItem = item
+                break
+            end
+        end
+    end
+
+    local augmentTable
+    if extData.decode(capeItem).augments then
+        augmentTable = extData.decode(capeItem).augments
+    end
+
+    local augValue
+    if augmentTable then
+        if string.contains(string.lower(pathItem.en), 'thread') then
+            augValue = augmentTable[threadIndex]
+        elseif string.contains(string.lower(pathItem.en),'dust') then
+            augValue = augmentTable[dustIndex]
+        elseif string.contains(string.lower(pathItem.en), 'dye') then
+            augValue = augmentTable[dyeIndex]
+        elseif string.contains(string.lower(pathItem.en), 'sap') then
+            augValue = augmentTable[sapIndex]
+        end
+    else
+        augValue = 'none'
+    end
+
+    if string.lower(augValue) == 'none' or not augmentTable then
+        return 'empty'
+    end
+
+    local max = maxAugMap[maxAugKey].max
+    if string.contains(augValue, max) then
+        windower.add_to_chat(123, 'You have augmented your ' .. currentCape.en .. ' to the max already with ' .. pathItem.en .. '.')
+        return 'maxed'
+    end
+
+    local mustContainTable = maxAugMap[maxAugKey].mustcontain
+    for k, augmentString in pairs(mustContainTable) do
+        if not string.contains(string.lower(augValue), string.lower(augmentString)) then
+            windower.add_to_chat(123,'You can\'t augment your ' .. currentCape.en .. ' with ' .. pathName .. ' because it has already been augmented with: ' .. augValue .. ' using ' .. pathItem.en .. '.')
+            return 'notmatching'
+        end
+    end
+
+    local cantContainTable = maxAugMap[maxAugKey].cantcontain
+    if table.length(cantContainTable) > 0 then
+        for k, augmentString in pairs(cantContainTable) do
+            if string.contains(string.lower(augValue), string.lower(augmentString)) then
+                windower.add_to_chat(123,'You can\'t augment your ' .. currentCape.en .. 'with ' .. pathName .. ' because it has already been augmented with: ' .. augValue .. ' using ' .. pathItem.en .. '.')
+                return 'notmatching'
+            end
+        end
+    end
+
+    return 'allclear'
+end
+
+function injectAugmentConfirmationPackets()
+    local optionIndex
+    if firstTimeAug then
+        firstTimeAug = false
+        optionIndex = 512
+    else
+        optionIndex = 256
+    end
+
+    local packet = packets.new('outgoing', 0x05B)
+    packet["Target"] = gorpaID
+    packet["Option Index"] = optionIndex
+    packet["_unknown1"] = pathIndex
+    packet["Target Index"] = gorpaTargetIndex
+    packet["Automated Message"] = true
+    packet["_unknown2"] = 0
```

Comment:
If you know what this is, see the comment above. If not and you're just setting it to `0` to be safe, you don't need to bother. All fields you don't explicitly specify are initialized to `0` by default.